test_gevent_server.py

```diff
@@ -0,0 +1,139 @@
+from multiprocessing import Process
+import pytest
+import __builtin__
+import gevent_server
+# import gevent_server
+import server
+import socket
+import os
+
+
+ADDR = ('127.0.0.1', 8000)
+_CRLF = b'\r\n'
+
+
+@pytest.yield_fixture()
+def server_setup():
+    process = Process(target=gevent_server.start_gserver)
+    process.daemon = True
+    process.start()
+    yield process
+
+
+@pytest.fixture()
+def client_setup():
+    client = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+    )
+    client.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    client.connect(ADDR)
+    return client
+
+
+def test_response_ok(server_setup):
+    test_body = b'body contents'
+    response = server.response_ok(test_body, 'text/html', len(test_body))
+    assert response == _CRLF.join([
+        b'HTTP/1.1 200 OK',
+        b'Content-Type: text/html',
+        b'Content-Length: 13',
+        test_body,
+        b''])
+
+
+def test_valid_content_type(monkeypatch):
+    monkeypatch.setattr(os.path, 'isdir', lambda x: False)
+    monkeypatch.setattr(os.path, 'abspath', lambda x: '/abs/path')
+    monkeypatch.setattr(os.path, 'dirname', lambda x: 'dirname')
+
+    # This class mocks out the opened file
+    class MockOpenResult(object):
+        def read(self):
+            return 'file.txt content 1 2 3'
+
+    # This class mocks out what happens with 'with open(filename) as f ...'
+    class MockOpenResultHelper(object):
+        def __enter__(self):
+            return MockOpenResult()
+
+        def __exit__(self, *args):
+            pass
+
+    monkeypatch.setattr(__builtin__, 'open', lambda x: MockOpenResultHelper())
+    (body, cont_type) = server.resolve_uri(b'/path/to/file.txt')
+    expected_body = b'file.txt content 1 2 3'
+    assert body == expected_body
+    assert cont_type == 'txt'
```

Comment:
Holy poop!  Mocking!  Nice.  We haven't even started to think about this, but good usage.